src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,120 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double COEFFICIENT = 1.5;
+    private static final int LIST_CAPACITY = 10;
+    private T[] data = (T[]) new Object[LIST_CAPACITY];
+    private int size;
+
     @Override
     public void add(T value) {
-
+        if (size == data.length) {
+            this.data = (T[]) grow();
+        }
+        data[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        isIndexInBoundAdd(index);
+        if (size == data.length) {
+            this.data = (T[]) grow();
+        }
+        Object[] data = this.data;
+        System.arraycopy(data, index, data, index + 1, size - index);
+        data[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list == null) {
+            return;
+        }
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        isIndexInBoundGet(index);
+        return data[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        isIndexInBoundGet(index);
+        data[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        isIndexInBoundGet(index);
+        final Object[] es = data;
+        T oldValue = (T) es[index];
+        fastRemove(es, index);
+        return oldValue;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        final Object[] es = data;
+        for (int i = 0; i < size; i++) {
+            if (isFulfillCondition(element, i)) {
+                T oldValue = (T) es[i];
+                fastRemove(es, i);
+                return oldValue;
+            }
+        }
+        throw new NoSuchElementException();
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
+    }
+
+    private Object[] grow(int minCapacity) {
+        int newCapacity = (int) Math.round(minCapacity * COEFFICIENT);
+        Object[] grown = new Object[newCapacity];
+        System.arraycopy(data, 0, grown, 0, data.length);
+        return grown;
+    }
+
+    private Object[] grow() {
+        return grow(size + 1);
     }
```

Comment:
```suggestion
```

Reply:
it is impossible it invokes inside growIfFull